vtex-apps/return-app
# Feat/invoice rma

#### What problem is this solving?

All the requests made with payment method `sameAsPurchase` will have the transaction automatically refunded - a invoice type Input will be created.

As now, the above behavior only happens when user selects to be refunded via credit card or the payment method was `sameAsPurchase` and the user has used a card to pay for the order. 

The requests are created with `sameAsPurchase` when the store sets `Return Payment Options` to same as Order in the settings for the app - as picture below.

<img width="463" alt="Screen Shot 2022-07-27 at 5 42 40 PM" src="https://user-images.githubusercontent.com/38737958/181368469-3aedc2a0-f732-45d7-acc4-3c2360e300a1.png">


#### How to test it?

For any request created with `sameAsPurchase`, move the status to `Refunded` in [this workspace](https://filainvoiceallrma--powerplanet.myvtex.com/).

After doing so, there is two ways to check the results:
1. Get the order information with a GET to `https://powerplanet.vtexcommercestable.com.br/api/oms/pvt/orders/:orderId`. On `packageAttachment.packages[number].type` search for the object with type Input. The value and items of that invoice should match the one approved via the App.

2. In the admin side, visit the transaction for that order. One should be able to find a refund with the same value approved in the app.
